satpy/etc/readers/hsaf_h5.yaml

```diff
@@ -0,0 +1,29 @@
+reader:
+  name: hsaf_h5
+  short_name: Hydrology SAF
+  long_name: Hydrology SAF products in HDF5 format
+  description: Reader for Hydrology SAF products
+  status: Beta, only h10 currently supported
+  supports_fsspec: false
+  reader: !!python/name:satpy.readers.yaml_reader.FileYAMLReader
+  sensors: [hsaf]
+
+file_types:
+  hsafh5:
+    file_reader: !!python/name:satpy.readers.hsaf_h5.HSAFFileHandler
+    file_patterns: ['S-HSAF-h10_{sensing_time:%Y%m%d}_day_merged.H5',
+                    'h10_{sensing_time:%Y%m%d}_day_merged.H5']
+datasets:
+  SC:
+    name: SC
+    msg_name: SC
+    sensor: hsaf
+    resolution: 3000
+    standard_name: seviri_snow_cover
+    units: class
```

Comment:
Not sure what the unit for categorical products should be. Currently none of the readers for categorical data has defined units. So lets leave it as `class` unless someone else has opinions on this.

Reply:
Sounds good to me. If someone has a better idea, we can easily adapt (or remove it altogether).

Reply:
nwcsaf reader sets units to "1" https://github.com/pytroll/satpy/blob/main/satpy/readers/nwcsaf_nc.py#L201-L202